### PR TITLE
Updated README.md with Xcode version bump

### PR DESCRIPTION
**Pull request by stechiu**
### Summary of changes

- Changed Xcode version requirements in README file as per Apple guidelines https://developer.apple.com/news/?id=khzvxn8a

### Checklist

- [ ] Added a changelog entry

- @stechiu

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

**Comment by scannillo** [comment]
We'll also want to update our 3 workflow files [here](https://github.com/braintree/braintree_ios/tree/main/.github/workflows) to use the latest Xcode version in our scripts. We'll probably have to bump our macOS version that the scripts run too in order to access the latest Xcode build. You can see them documented here - https://github.com/actions/runner-images/tree/main/images/macos

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Welcome to Braintree's iOS SDK. This library will help you accept card and alter
 
 v6 is the latest major version of Braintree iOS. To update from v5, see the [v6 migration guide](https://github.com/braintree/braintree_ios/blob/main/V6_MIGRATION.md). If you have not yet migrated to v5, see the [v5 migration guide](https://github.com/braintree/braintree_ios/blob/5.x/V5_MIGRATION.md)
 
-**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 14.3+ and Swift 5.8+.
+**The Braintree iOS SDK permits a deployment target of iOS 15.0 or higher**. It requires Xcode 15.0+ and Swift 5.8+.
```
Comment (jaxdesmarais) [inline README.md:13]:
```suggestion
**The Braintree iOS SDK permits a deployment target of iOS 14.0 or higher**. It requires Xcode 15.0+ and Swift 5.8+.
```

Comment (stechiu) [reply README.md:13]: Thanks!

**Comment by stechiu** [comment]
> We'll also want to update our 3 workflow files [here](https://github.com/braintree/braintree_ios/tree/main/.github/workflows) to use the latest Xcode version in our scripts. We'll probably have to bump our macOS version that the scripts run too in order to access the latest Xcode build. You can see them documented here - https://github.com/actions/runner-images/tree/main/images/macos

Thanks! I've updated the script in this PR https://github.com/braintree/braintree_ios/pull/1215

**Pull request merged**